### PR TITLE
Sagemaker Endpoint LLM

### PR DESCRIPTION
**Pull request by 3coins**
Updates #965

### CODE REVIEW

**Comment by jainsid24** [comment]
@3coins how to provide a model name in this? Say I create an Sagemaker endpoint config with an LLM, how should I add that parameter when calling SagemakerEndpoint to chose the model?

File: docs/modules/document_loaders/examples/sagemaker.ipynb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Dict\n",
+    "\n",
+    "from langchain import PromptTemplate, SagemakerEndpoint, ContentHandlerBase\n",
```
Comment (jainsid24) [inline docs/modules/document_loaders/examples/sagemaker.ipynb:48]: ContentHandlerBase can't be import this way, this works ⬇️ 
```from langchain.llms.sagemaker_endpoint import ContentHandlerBase```

Comment (3coins) [reply docs/modules/document_loaders/examples/sagemaker.ipynb:48]: @jainsid24 
Thanks for pointing out these points. Let me spend some more time today on this, will post an update soon.

**Comment by 3coins** [comment]
@jainsid24 
Fixed references in the notebook, also added a new optional dict as input where `TargetModel` and other arguments that `invoke_endpoint` supports can be passed. Thanks for reviewing this change.

**Comment by jainsid24** [comment]
@3coins thanks for adding the changes. This is now working as expected. I've been trying to use this on [Flan-T5-XXL](https://huggingface.co/google/flan-t5-xxl) model and it works flawlessly. You can add this optional section to the notebook as well for those who are using `Flan-T5` model.

```
class ContentHandler(ContentHandlerBase):
    content_type = "application/json"
    accepts = "application/json"

    def transform_input(self, prompt: str, model_kwargs: Dict) -> bytes:
        input_data = {"text_inputs": [prompt], **model_kwargs}
        input_str = json.dumps(input_data)
        return input_str.encode('utf-8')
    
    def transform_output(self, output: bytes) -> str:
        response_json = json.loads(output.read().decode("utf-8"))
        return response_json["generated_texts"][0]

content_handler = ContentHandler()

parameters = {
  "early_stopping": True,
  "length_penalty": 2.0,
  "temperature": 0,
  "min_length": 10,
  "no_repeat_ngram_size": 3,
}

chain = load_qa_chain(
    llm=SagemakerEndpoint(
        endpoint_name="jumpstart-dft-hf-text2text-flan-t5-xl", 
        credentials_profile_name="default", 
        region_name="us-west-2", 
        model_kwargs=parameters,
        content_handler=content_handler
    ),
    prompt=PROMPT
)

chain({"input_documents": docs, "question": query}, return_only_outputs=True)
```

@hwchase17 We will use this for production usecases, have tested this. Would be great if we can fast-pace this PR :)

**Comment by hwchase17** [comment]
awesome! fast-tracked for release tmrw - thanks @3coins and @jainsid24 !

**Review by hwchase17 — APPROVED** [review]

**Comment by 3coins** [comment]
@hwchase17 
Thanks for merging.

**Pull request merged**